### DISCUSSION TITLE
parser request body content-type text/xml

### DISCUSSION THREAD

**Discussion by chrishj59** [vercel/next.js#40795] [Help]
Hi - 
In my API route a request body is passed from PrimeReact fileUpload as text/xml.

Could you let me know how to add XML parser for the route to decode the request body?

This is what I get if I console.log @body

```
------WebKitFormBoundarycJWsWYXYPjl7HRb2
Content-Disposition: form-data; name="edcStock"; filename="edc_xml_en_test.xml"
Content-Type: text/xml


<?xml version="1.0" encoding="UTF-8"?>
<products>
  <product>
    <id>32</id>

```

Here is the API route
```
export default function handler(req: NextApiRequest, res: NextApiResponse) {
	const { body } = req;
	console.log(body);
	return res.status(200).json({ message: 'called api' });
}
```

**Comment by icyJoseph** [comment]
Ah this question again. 

First you gotta turn off the body parser:

```ts
// add this export to the API route file
export const config = {
  api: {
    bodyParser: false,
  },
}

export default function handler(req: NextApiRequest, res: NextApiResponse) {
	const { body } = req;
	console.log(body);
	return res.status(200).json({ message: 'called api' });
}
```

Now the body is a stream of data to parse. From there you've got a few options. You can try to use [Formidable](https://www.npmjs.com/package/formidable), for example.

The key thing here is that the data you are receiving is not `application/json` anymore, but `multipart/form-data`. More on that here, https://developer.mozilla.org/en-US/docs/Learn/Forms/Sending_and_retrieving_form_data#a_special_case_sending_files -

And well, you gotta pipe that to memory or to a file inside your API route. 

## EDIT

I have a snippet I've shared around on this discussions forum, but before I copy pasted that, I found this article that perhaps can help you out, it has snippets too: https://chadalen.com/blog/how-to-use-a-multipart-form-in-nextjs-using-api-routes
